xarray/core/dataarray.py

```diff
@@ -816,6 +816,118 @@ def swap_dims(self, dims_dict):
         ds = self._to_temp_dataset().swap_dims(dims_dict)
         return self._from_temp_dataset(ds)
 
+    def set_index(self, indexers=None, append=False, inplace=False,
```

Comment:
A note on terminology: in xarray/pandas, `indexer` usually refers the argument/key used for indexing, whereas `index` refers to the set of existing labels, e.g., `df.index` vs `df.loc[indexer]`. So in this case I think the argument name `indexes` would make more sense.